src/main/scala/nl/knaw/dans/api/sword2/DepositHandler.scala

```diff
@@ -152,51 +154,117 @@ object DepositHandler {
     }
   }
 
-  def resolveFetchItems(bagitDir: File)(implicit id: String): Try[Unit] = Try {
-    log.debug(s"[$id] Checking if bag contains fetch.txt")
-    getFetchTxt(bagitDir).foreach(fetchText =>
-      { checkFetchTxtUrls(fetchText)
-        resolveFetchTxtFiles(bagitDir, fetchText)
-      }
-    )
-  }
-
   private def getFetchTxt(bagitDir: File)(implicit id: String): Option[FetchTxt] = Option {
     getBagFromDir(bagitDir).getFetchTxt
   }
 
-  private def checkFetchTxtUrls(fetchText: FetchTxt)(implicit id: String): Unit = {
+  def checkFetchTxtUrls(bagitDir: File)(implicit id: String): Try[Unit] = Try {
     log.debug(s"[$id] Checking validity of urls in fetch.txt")
-    fetchText.asScala.foreach(item =>
-      checkUrlValidity(item.getUrl))
+    getFetchTxt(bagitDir).foreach(fetchText =>
+      fetchText.asScala.foreach(item =>
+      checkUrlValidity(item.getUrl)))
   }
 
   private def checkUrlValidity(url: String)(implicit id: String): Unit = {
     // check if the url is syntactically correct
     try {
       new URL(url)
     } catch {
-      case e: MalformedURLException => throw new InvalidDepositException(id, s"Invalid url in Fetch Items ($url)")
+      case e: MalformedURLException => throw InvalidDepositException(id, s"Invalid url in Fetch Items ($url)")
     }
     // check if the url complies with the allowed url-structure
     val urlPattern = Pattern.compile(SwordProps("url-pattern"))
     if (!urlPattern.matcher(url).matches)
-      throw new InvalidDepositException(id, s"Not allowed url in Fetch Items ($url)")
+      throw InvalidDepositException(id, s"Not allowed url in Fetch Items ($url)")
   }
 
-  private def resolveFetchTxtFiles(bagitDir: File, fetchText: FetchTxt)(implicit id: String): Unit = {
-    log.debug(s"[$id] Resolving files in fetch.txt")
-    fetchText.asScala.foreach(item =>
-      Using.urlInputStream(new URL(item.getUrl)).foreach(src =>
-        Files.copy(src, Paths.get(bagitDir.getAbsolutePath, item.getFilename))))
-  }
-
-  def checkBagValidity(bagitDir: File)(implicit id: String): Try[Unit] = {
+  def checkBagValidity(bagitDir: File)(implicit id: String) = {
```

Comment:
You didn't for `checkBagValidity`. No return type. However, if you generate it automagically, it will give you `Try[Any]`. That's because you have a `Success(Unit)` in your code. This is incorrect. Let me explain this:

`Unit` is a **type**, just like `Boolean`, `Int`, `String`, etc. `Unit` has only one value, namely `()`. The confusing bit is that you pronounce this as *unit* as well. So, you have to write `()` rather than `Unit`.

If you change this, you will get the correct return type, which is `Try[Unit]`.